src/modules/p_usrloc/ul_db.c

```diff
@@ -102,13 +102,16 @@ int ul_db_child_init(void) {
 	if(ul_db_child_locnr_init() == -1) return -1;
 	
 	LM_INFO("location number is %d\n", max_loc_nr);
-	if(db_master_write){
+        lock_get(&write_on_master_db_shared->lock);
```

Comment:
Check whether in ul_db_init() you need to acquire locks at all. It seems like this function is called one time only, before any fork is done, so most probably no synchronisation is needed. Same for lock release.

Reply:
p_usrloc_mod.c:
The fork() system call is triggered in the init_db_check() function, which takes place just before calling the ul_db_child_init() function where the above mentioned lock checking mechanism is used.